filters/alpha_ratio.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+"""Filters the lines based on the ratio between alphabetic characters in a line from the language and others"""
+from sys import stdin, stdout, stderr
+from typing import Optional
+import argparse
+import re
+from clean_common import CHARS
+
+FILTER_PARAM = { "filter_alpha_ratio": {
```

Comment:
I like these options, but I'm conflicted about how you do `allowed_values` here. I want to support this, it's pretty essential. Just don't know how yet.

I want to avoid parsing external python files just to figure out all available filters. Like, I'm not sure how that would work out when you'd have more complex filters like bicleaner which might be installed in their on virtualenv with their own dependencies. It's pretty easy to make them callable from outside. Hell, if all else fails, you could wrap them in a shell script that loads the proper environment etc.

Most of the parameters can also just be described in a json file. That would not lock us into python, and versioning json schemas is a lot easier than maintaining different versions of some python API. However, options like the `available_options` key, which are dynamic here, wouldn't work.

One work-around for that could be to allow a `$(...)` sort of syntax that could be used to ask the filter to print all available options. You could then do something like:

```json
{
"allowed_values": "$(filters/alpha_ratio.py --help | sed -n 's/^ *--src-lang {\(.*\)}/\1/p' | tr ',' '\n')"
}
```

… but looking at that example, it looks pretty unwieldy.

Another one, if most filter tools use argparse anyway, would be to just parse the output of `filter.py --help`. It looks pretty standardised and parsable. Only downside I see right now is that there is no way to differentiate between str and float arguments.

Two other problems are: boolean and optional parameters.

Flags, or boolean arguments like `--debug` won't work with the simple variable replacement method used now.

Same for optional parameters. How do I specify no value for `--src-lang` so it falls back to the default? Leaving `$LANG1` blank won't remove the `--src-lang` bit from the command.